### PR TITLE
conditionally alter the regex for determining locale parts. fixes #1

### LINKED ISSUE #1: Build task on windows fails due to OS path separator issue
The grunt-makara-amdify 1.0.1 uses the makara-builder 1.0.0 and in makara-builder/index.js I can see:

```
var locales = paths.map(function (p) {
    var m = /(.*)\/(.*)/.exec(path.relative(localeRoot, p));
    return m[2] + '-' + m[1];
});
```

localeRoot="...\locales" and p="...locales/DE/de"

The path.relative() call transforms p into windows style and returns "DE\de".
The reg ex evaluation returns null because of the hardcoded linux path sep.
The following line gives of course "Fatal error: Cannot read property '2' of null".

Any hint for a workaround? Beside "use linux" ... ;-)

### PR DESCRIPTION
**Pull request by grawk**
- conditionally alter the regex for determining locale parts. fixes #1
- add unit testing
- add travis integration

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -3,6 +3,8 @@
 var path = require('path');
 var glob = require('glob');
 var async = require('async');
+var os = require('os');
+var re = (os.platform() === 'win32') ? /(.*)\\(.*)/ : /(.*)\/(.*)/;
```
Comment (jasisk) [inline index.js]: y u no `path.sep`?

File: test/index.js [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+tap.test('check that locale directories are properly identified', function (t) {
+    t.plan(3);
+    t.type(builder, 'function');
+    t.equal(builder.length, 3);
+    builder(appRoot, writer, function (err) {
+        if (err) {
+            throw err;
+        }
+        t.equal(locales.length, 0);
+        t.end();
```
Comment (jasisk) [inline test/index.js]: Superfluous. You have a `plan`.

**Comment by grawk** [comment]
OK how's that @jasisk ?

**Comment by skoranga** [comment]
looks good.

**Pull request merged**